services/core-web/src/actionCreators/mineAlertActionCreator.ts

```diff
@@ -0,0 +1,105 @@
+import { notification } from "antd";
+import { showLoading, hideLoading } from "react-redux-loading-bar";
+import { ENVIRONMENT, IMineAlert } from "@mds/common";
+import { error, request, success } from "@common/actions/genericActions";
+import { createRequestHeader } from "@common/utils/RequestHeaders";
+import CustomAxios from "@common/customAxios";
+import * as API from "@/constants/API";
+import * as mineActions from "@/actions/mineAlertActions";
+import * as reducerTypes from "@/constants/reducerTypes";
+import { AxiosResponse } from "axios";
+import { AppThunk } from "@/store/appThunk.type";
+
+export const fetchMineAlertsByMine =
+  (mineGuid: string): AppThunk<Promise<AxiosResponse<IMineAlert>>> =>
+  (dispatch): Promise<AxiosResponse<IMineAlert>> => {
+    dispatch(request(reducerTypes.GET_MINE_ALERTS));
+    dispatch(showLoading("modal"));
+    return CustomAxios()
+      .get(`${ENVIRONMENT.apiUrl}${API.MINE_ALERTS(mineGuid)}`, createRequestHeader())
+      .then((response: AxiosResponse<IMineAlert>) => {
+        dispatch(success(reducerTypes.GET_MINE_ALERTS));
+        dispatch(mineActions.storeMineAlerts(response.data));
+        return response;
+      })
+      .catch((err) => {
+        dispatch(error(reducerTypes.GET_MINE_ALERTS));
+        throw new Error(err);
+      })
+      .finally(() => dispatch(hideLoading()));
+  };
+
+export const createMineAlert =
+  (mineGuid: string, payload: IMineAlert): AppThunk<Promise<AxiosResponse<IMineAlert>>> =>
```

Comment:
Was just chatting with Alex.  He had a good catch related to his PR.  I think the payload and the response will need their own interfaces as the data structure isn't the same.

Reply:
Nice catch. I have updated my PR to fix this.